database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -44,9 +41,8 @@ public FirebaseIndexArray(Query keyRef, Query dataRef) {
     @Override
     public void cleanup() {
         super.cleanup();
-        Set<Query> refs = new HashSet<>(mRefs.keySet());
-        for (Query ref : refs) {
-            ref.removeEventListener(mRefs.remove(ref));
+        for (Map.Entry<Query, ValueEventListener> entry : mRefs.entrySet()) {
+            entry.getKey().removeEventListener(entry.getValue());
```

Comment:
This is not equivalent to the previous behavior because there is no call to `mRefs.remove()`